components/chat/chatBar/CatalogListBox.tsx

```diff
@@ -50,25 +45,27 @@ export const CatalogListBox = forwardRef<ToolCatalogRef, CatalogListboxProps>(
     const debouncedQuery = useDebouncedValue(query || '', 250);
 
     const featuredResults = useMemo(() => {
-      const flattened = (featuredTools: FeaturedTools) =>
-        Object.entries(featuredTools).flatMap(([category, tools]) =>
-          tools.map((tool) => ({ tool, category }))
-        );
+      if (debouncedQuery.trim().length <= 0)
```

Comment:
nit: I would like to see this DRY-ed up, if we can. It is repeated in `catalog.tsx` too.